src/with_generic_array.rs

```diff
@@ -111,26 +149,37 @@ impl<T: 'static + Default, Cap: ArrayLength<T>> RingBuffer<T> for GenericRingBuf
     #[inline]
     fn push(&mut self, value: T) {
         if self.is_full() {
+            unsafe {
+                // make sure we drop whatever is being overwritten
+                // SAFETY: the buffer is full, so this must be inited
+                //       : also, index has been masked
+                let index = crate::mask(self, self.readptr);
```

Comment:
could you maybe place the masking outside the unsafe block, so as little code as possible is marked unsafe?